src/Nox.Types/Types/StreetAddress/StreetAddress.cs

```diff
@@ -1,10 +1,213 @@
+using System.Linq;
+using System.Text.RegularExpressions;
+
 namespace Nox.Types;
 
-    /// <summary>
-    /// Represents a Nox <see cref="StreetAddress"/> type and value object.
-    /// </summary>
-    /// <remarks>Placeholder, needs to be implemented</remarks>
-    public sealed class StreetAddress : ValueObject<(string Line1,string Line2, string Line3,string ZipCode,string City, CountryCode2 CountryCode2), StreetAddress>
+/// <summary>
+/// Represents a Nox <see cref="StreetAddress"/> type and value object.
+/// </summary>
+/// <remarks>Placeholder, needs to be implemented</remarks>
+public sealed class StreetAddress : ValueObject<
+    (int StreetNumber,
+    string AddressLine1,
+    string AddressLine2,
+    string Route,
+    string Locality,
+    string Neighborhood,
+    string AdministrativeArea1,
+    string AdministrativeArea2,
+    string PostalCode,
+    CountryCode2 CountryId), StreetAddress>
+{
+    private readonly Regex _postalCodeRegex = new("^\\d{5}(?:[-\\s]\\d{4})?$");
```

Comment:
this regex depends on the Country (CountryCode), we need to look to Nox.Reference and see if we can bring some code or use something like this https://gist.github.com/jamesbar2/1c677c22df8f21e869cca7e439fc3f5b

Reply:
There is a possibility to determine postal code format and pattern by country.
But I suppose Nox.Type won't refer on Nox.Reference (?)

As for using the same code. Nox.Reference uses data import to store these data in sqllite db, 
We cann't use this code in Nox.Type since it invokes external resources. which is overhead for Nox.Types. May be we can think about generation kind of constants for this porposes or just use constants for each country.

We should keep in mind that address can be missed. Therefore need to care about default value.

Reply:
Should not reference Nox.References 

We need to hold this data/configuration in Nox.Types without any external dependency

Reply:
Provided CountryPostalCodeValidator class which holds mapping between country and postal code patterns